internal/builders/gradle/README.md

```diff
@@ -0,0 +1,121 @@
+# Generation of SLSA3+ provenance for `.jar` artifacts built with Gradle
+
+This document explains how to generate SLSA provenance for artifacts built with Gradle.
+
+This can be done by adding a step to your Github Actions workflow to call a
+[reusable
+workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows)
+to build the package and generate SLSA provenance. We'll call this
+workflow the "Gradle builder" from now on.
+
+---
+
+<!-- markdown-toc --bullets="-" -i README.md -->
+
+<!-- toc -->
+
+- [Benefits of Provenance](#benefits-of-provenance)
+- [Development status](#development-status)
+- [Limitations](#limitations)
+- [Generating Provenance](#generating-provenance)
+  - [Getting Started](#getting-started)
+  - [Private Repositories](#private-repositories)
+- [Verification](#verification)
+
+<!-- tocstop -->
+
+---
+
+## Benefits of Provenance
+
+Using the Gradle builder will generate a non-forgeable attestation to the
+Gradle package using the identity of the GitHub workflow. This can be used to
+create a positive attestation to a package coming from your repository.
+
+That means that once your users verify the package they have downloaded they can
+be sure that it was created by your repository's workflow and hasn't been
+tampered with.
+
+## Development status
+
+The Gradle builder is currently in alpha. The API could change while approaching
+a Generally Available (GA) release. You can track progress towards General
+Availability via
+[this milestone](https://github.com/slsa-framework/slsa-github-generator/milestone/17).
+
+Please try it out and
+[create an issue](https://github.com/slsa-framework/slsa-github-generator/issues/new)
+to send us feedback!
+
+## Limitations
+
+The Gradle builder currently has the following limitations:
+
+1. The project must be build'able by way of `gradle build`. If you need the option for flags, profiles or something else to define more granular builds, please open an issue.
+
+## Generating Provenance
+
+The Gradle builder uses a Github Actions reusable workflow to build your
+package and generate the provenance.
+
+### Getting Started
+
+Let's say you have the following build set up:
+
+1. You can build your artifacts by way of `gradle build`.
+2. You release artifacts via Github Actions.
+
+To add provenance to releases is easy. Simply use the following workflow in `.github/workflows` in your repository:
+
+```yaml
+name: SLSA Provenance with Gradle builder
+on:
+  - workflow_dispatch
+
+permissions: read-all
+
+jobs:
+  usetrw:
+    permissions:
+      id-token: write
+      contents: read
+      actions: read
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_gradle_slsa3.yml@v1.7.0
+    with:
+      rekor-log-public: true
```

Comment:
nit: I'd like to leave this off examples since folks are likely to copy and paste them, and we would really like them to understand what this option does before setting it (we already have docs in the Private Repositories section).

Reply:
Let me send a PR to remove these in all examples. Good catch

Reply:
https://github.com/slsa-framework/slsa-github-generator/pull/2466